verifier/verifier.go

```diff
@@ -135,18 +135,27 @@ func (v *verifier) Verify(ctx context.Context, desc ocispec.Descriptor, signatur
 	return outcome, outcome.Error
 }
 
-func (v *verifier) processSignature(ctx context.Context, sigBlob []byte, envelopeMediaType string, trustPolicy *trustpolicy.TrustPolicy, pluginConfig map[string]string, outcome *notation.VerificationOutcome) error {
+func (v *verifier) processSignature(ctx context.Context, sigBlob []byte, envelopeMediaType string, trustPolicy *trustpolicy.TrustPolicy, pluginConfig map[string]string, outcome *notation.VerificationOutcome, userMetadata map[string]string) error {
 	logger := log.GetLogger(ctx)
 
 	// verify integrity first. notation will always verify integrity no matter what the signing scheme is
-	envContent, integrityResult := verifyIntegrity(sigBlob, envelopeMediaType, outcome)
+	envContent, integrityResult := verifyIntegrity(ctx, sigBlob, envelopeMediaType, outcome)
 	outcome.EnvelopeContent = envContent
 	outcome.VerificationResults = append(outcome.VerificationResults, integrityResult)
 	if integrityResult.Error != nil {
 		logVerificationResult(logger, integrityResult)
 		return integrityResult.Error
 	}
 
+	// verify user metadata
+	if len(userMetadata) > 0 {
+		logger.Info("User Metadata flag is present. Checking signature metadata for specified values.")
+		err := verifyUserMetadata(logger, envContent, userMetadata)
```

Comment:
Should this happen after below code? https://github.com/notaryproject/notation-go/blob/6c88d3de1283d0d573bafae9d1567026367bfcb6/verifier/verifier.go#L130-L134?

In that way we can avoid parsing payload twice and only perform annotation check if descriptor is correct